crates/relayer/src/chain/cosmos/query/tx.rs

```diff
@@ -183,7 +183,31 @@ pub async fn query_packets_from_block(
         }
     }
 
-    Ok(result)
+    begin_block_events.append(
+        &mut block_results
+            .begin_block_events
+            .unwrap_or_default()
+            .into_iter()
+            .filter_map(|ev| filter_matching_event(ev, request, &request.sequences))
+            .map(|ev| IbcEventWithHeight::new(ev, height))
+            .collect(),
+    );
+
+    end_block_events.append(
+        &mut block_results
+            .end_block_events
+            .unwrap_or_default()
+            .into_iter()
+            .filter_map(|ev| filter_matching_event(ev, request, &request.sequences))
+            .map(|ev| IbcEventWithHeight::new(ev, height))
+            .collect(),
+    );
+
+    let mut events = begin_block_events;
+    events.append(&mut tx_events);
+    events.append(&mut end_block_events);
+
+    Ok(events)
```

Comment:
Implementation note: Do I understood correctly that this changes here is strictly for #2868 ?